src/GoogleFit/FitOnFhir.GoogleFit/Services/UsersService.cs

```diff
@@ -144,5 +152,20 @@ public override async Task RevokeAccessRequest(PlatformUserInfo platformUserInfo
                 platformUserInfo.ImportState = DataImportState.Unauthorized;
             }
         }
+
+        private async Task RedirectAuthorization(AuthState state, CancellationToken cancellationToken)
+        {
+            var query = new Dictionary<string, string>()
+            {
+                [Constants.StateQueryParameter] = state.State,
+            };
+            var uri = QueryHelpers.AddQueryString(state.RedirectUrl, query);
```

Comment:
This can be simplified a bit, instead of creating a dictionary:

```c#
var uri = QueryHelpers.AddQueryString(state.RedirectUrl, Constants.StateQueryParameter, state.State);
```

Reply:
Actually. You might need to check if the 'state' is null before doing this. I think it makes sense that if the caller did not include a state string in the request, we should not include it in the response.